builtin/providers/google/resource_compute_router_peer_test.go

```diff
@@ -0,0 +1,263 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+
+	"google.golang.org/api/compute/v1"
+)
+
+func TestAccComputeRouterPeer_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeRouterPeer_basic,
+				Check: testAccCheckComputeRouterPeerExists(
+					"google_compute_router_peer.foobar"),
+			},
+			resource.TestStep{
+				Config: testAccComputeRouterPeer_keepRouter,
+				Check: testAccCheckComputeRouterPeerDestroy(
+					"google_compute_router_peer.foobar"),
+			},
+		},
+	})
+}
+
+func testAccCheckComputeRouterPeerDestroy(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+		project := config.Project
+
+		routersService := compute.NewRoutersService(config.clientCompute)
+
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != "google_compute_router_peer" {
+				continue
+			}
+
+			region := rs.Primary.Attributes["region"]
```

Comment:
Would love to see this support the `getRegion` checks the same way the resource does, to avoid nasty bugs down the road.